src/amenities/amenities.service.ts

```diff
@@ -46,7 +69,9 @@ export class AmenitiesService {
     }
   }
 
-  async addAmenities(id: string, dto: AmenitiesDto) {
+  async addAmenities(id: string, dto: AmenitiesDto, userId: string) {
+    const isOwner = await this.verifyOwner(userId, id);
+    if (!isOwner) throw new UnauthorizedException('You are not authorized to perform this action');
```

Comment:
And now throw NotFoundException('Accomodation with this id and ownerId are not found'), with the message like this.